ta/pkcs11/src/pkcs11_token.c

```diff
@@ -1464,3 +1464,85 @@ enum pkcs11_rc entry_ck_logout(struct pkcs11_client *client,
 
 	return PKCS11_CKR_OK;
 }
+
+enum pkcs11_rc entry_ck_seed_random(struct pkcs11_client *client,
+				    uint32_t ptypes, TEE_Param *params)
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INOUT,
+						TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	TEE_Param *ctrl = params;
+	TEE_Param *in = params + 1;
+	enum pkcs11_rc rc = PKCS11_CKR_OK;
+	struct serialargs ctrlargs = { };
+	struct pkcs11_session *session = NULL;
+
+	if (!client || ptypes != exp_pt)
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	serialargs_init(&ctrlargs, ctrl->memref.buffer, ctrl->memref.size);
+
+	rc = serialargs_get_session_from_handle(&ctrlargs, client, &session);
+	if (rc)
+		return rc;
+
+	if (serialargs_remaining_bytes(&ctrlargs))
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	if (in->memref.size && !in->memref.buffer)
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	/* noop - just ignore the input seed */
```

Comment:
I see an error code "CKR_RANDOM_SEED_NOT_SUPPORTED" in teh specification. Since TEE doesn't support seeding the random number generator, should we return "CKR_RANDOM_SEED_NOT_SUPPORTED".

Reply:
`TEE_GenerateRandom()` is based on core internal `crypto_rng_read()` that can be seeded with system PTA command `PTA_SYSTEM_ADD_RNG_ENTROPY`. Should we use it here?

Reply:
Shall it be used is a good question -- it does provide access adjust the state of PRNG which I cannot say is it a good idea or not -- someone with better knowledge of PRNG in use should give advice in here.

I can try to see what kind of code that would be to call PTA method.

Reply:
About return value -- I decided just to fake it as success not to cause any hardship for the caller. I don't think REE should have any visibility how the RNG functionality operates.

Reply:
Added [review] commit with the functionality.

Reply:
merged to other commits.